src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -238,13 +245,23 @@ private static SafeFileHandle CreateSharedBackingObject(Interop.Sys.MemoryMapped
 
             static string GenerateMapName()
             {
-                const int MaxSharedMemoryObjectNameLength = 32; // SHM_NAME_MAX on OSX ARM64, on other systems it's equal PATH_MAX (250)
-                // The POSIX shared memory object name must begin with '/'.  After that we just want something short (32) and unique.
-                return string.Create(MaxSharedMemoryObjectNameLength, 0, (span, state) =>
+                // macOS shm_open documentation says that the sys-call can fail with ENAMETOOLONG if the name exceeds SHM_NAME_MAX characters.
+                // The problem is that SHM_NAME_MAX is not defined anywhere and is not consistent amongst macOS versions (arm64 vs x64 for example).
+                // It was reported in 2008 (https://lists.apple.com/archives/xcode-users/2008/Apr/msg00523.html),
+                // but considered to be by design (http://web.archive.org/web/20140109200632/http://lists.apple.com/archives/darwin-development/2003/Mar/msg00244.html).
+                // According to https://github.com/qt/qtbase/blob/1ed449e168af133184633d174fd7339a13d1d595/src/corelib/kernel/qsharedmemory.cpp#L53-L56 the actual value is 30.
+                // Some other OSS libs use 32 (we did as well, but it was not enough) or 31, but we prefer 30 just to be extra safe.
+                const int MaxNameLength = 30;
+                // The POSIX shared memory object name must begin with '/'.  After that we just want something short (30) and unique.
+                const string NamePrefix = "/dotnet_";
+                return string.Create(MaxNameLength, 0, (span, state) =>
```

Comment:
Although it comes at the cost of a copy, it looks like:

```c#
        return $"/dotnet_{Guid.NewGuid()}".Substring(0, MaxNameLength);
```
has a better [codegen](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBuIGYACMxgYRoG8bGfmnjyABkYBZcgAoAlN15dqvBYwBu2KIwB22fDEYBeRgBIARAHoAJhAzqYGAPocA4gFcAlmYB0AORgB3Z26kAXyMAbhlFZgB2DS0YdwBlJ2ABQXFBNEZ6QUkw+V5AmnC+ZiFRUikiuQieSHVcDEYXdQaRbARPWIAZGHUAcwwACz1MwVzqxlr6kuEO7QAFWAAzFwRh0wsrG1tQooViaJT3VlhsDBhxVvaunv6BjPSS0lZgHMK8iOIUR+fxeIAHbDqAA8YAGqgAfIxcAD1Bkmg16qcYNJ3ooquNeP9ASCwVBIb1XGZhoiwABrbAAGwpEDAE1xAG16KQALpjDE8fwebx+QlSdwAFSgAE8AGLQfCncQEtwZCBOBrwumqXAAdSgLgwZ1hjCMniMr1R41mMAWMGWCCOED+Qv5EHE0MBBvZPGlHniFJcYHOD0uxu6fUGjDgjGNpvN7n9t0kluttvtMISHq94lDSxWEZug0kTuqBVRecCQA) and easier to read.

Reply:
In the sharplab this doesn't use the interpolated string-handlers, which would be used here.
Then the codegen looks different, and the current approach better.

Reply:
Interesting. I haven't compared it with`DOTNET_JitDisasm`, but if you have, I'll take your word for it.
I still think if it isn't a hot _enough_ path, simplicity is better.

Reply:
Could also be written as 
```c#
const int MaxNameLength = 30;
const string NamePrefix = "/dotnet_";
Span<char> guid = stackalloc char[32];
Guid.NewGuid().TryFormat(guid, out int charsWritten, "N");
Debug.Assert(charsWritten == 32);
return $"{NamePrefix}{guid[..(MaxNameLength - NamePrefix.Length)]}";
```

Besides that: must it be based on Guid? We just need something random (of length 22).
Are different random enough thing that can be easily created?

> I still think if it isn't a hot enough path, simplicity is better.

For sure.